src/matrices.jl

```diff
@@ -293,13 +340,14 @@ function recurrence_matrix(x::AbstractVector, y::AbstractVector, metric, ε, par
     return sparse(rowvals, colvals, nzvals, length(x), length(y))
 end
 
-function recurrence_matrix(x::AbstractVector, metric::Metric, ε::Real, parallel::Val{false})
+function recurrence_matrix(x::AbstractVector, metric::Metric, ε::Vector, parallel::Val{false}, fan::Val{false})
+    @assert length(ε) == length(x)
     rowvals = Vector{Int}()
     colvals = Vector{Int}()
     for j in 1:length(x)
         nzcol = 0
         for i in 1:j
-            if @inbounds abs(x[i] - x[j]) ≤ ε
+            if @inbounds evaluate(metric, x[i], x[j]) ≤ ε[i]
```

Comment:
I do not understand the change in line 350: If `x` is a vector, all metrics should give the same result as `abs`, isn't it? Perhaps it's an accidental copy-and-paste from another method. It also happens in the next method, when the FAN method is applied.

Reply:
yeap you are right, these two lines coincide when `x` is a vector